runtime/tchannel_client.go

```diff
@@ -219,8 +219,14 @@ func (c *tchannelOutboundCall) finish(err error) {
 }
 
 func (c *tchannelOutboundCall) logFields() []zapcore.Field {
+	var hostPort string
+	if c.call != nil {
+		hostPort = c.call.RemotePeer().HostPort
+	} else {
+		hostPort = ""
+	}
 	fields := []zapcore.Field{
-		zap.String("remoteAddr", c.call.RemotePeer().HostPort),
+		zap.String("remoteAddr", hostPort),
```

Comment:
do we still need to log `hostPort` if it's empty string?
I'd suggest just create a list of fields and append `hostPort` if it's not null